drivers/default.go

```diff
@@ -0,0 +1,90 @@
+package drivers
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+// DefaultDriver - Embed this struct into drivers to provide the common set
+// of fields and functions.
+type DefaultDriver struct {
+	storePath      string
+	IPAddress      string
+	SSHUser        string
+	SSHPort        int
+	MachineName    string
+	CaCertPath     string
+	PrivateKeyPath string
+	SwarmMaster    bool
+	SwarmHost      string
+	SwarmDiscovery string
+}
+
+// NewDefaultDriver - Get an instance of a DefaultDriver
+func NewDefaultDriver(machineName string, storePath string, caCert string, privateKey string) *DefaultDriver {
+	return &DefaultDriver{
+		MachineName:    machineName,
+		storePath:      storePath,
+		CaCertPath:     caCert,
+		PrivateKeyPath: privateKey,
+	}
+}
+
+// GetSSHKeyPath -
+func (d *DefaultDriver) GetSSHKeyPath() string {
+	return d.GetStorePath("id_rsa")
+}
+
+// GetStorePath -
+func (d *DefaultDriver) GetStorePath(path string) string {
```

Comment:
I don't like the name of this function, but I couldn't think of anything better at the time... Now I'm thinking something like `ResolveStorePath`... any thoughts?